docs/integration-guide.md

```diff
@@ -107,6 +107,239 @@ Here's how we recommend doing it:
 
+### Files that need to be updated
+
+* `templates/zerver/integrations.html`: Edit to add end-user documentation and
+  integration icon. See [Documenting your
+  integration](#documenting-your-integration) for details.
+* `zerver/test_hooks.py`: Edit to include tests for your webbook. See [Testing
+  and writing tests](testing.html) for details.
+* `zproject/urls.py`: Edit to add externally available url of the webhook and
+  associate with the function added to `zerver/views/webhooks/mywebhook.py`
+
+### Files that need to be created
+
+Select a name for your webhook and use it consistently. The examples below are
+for a webhook named 'mywebhook'.
+
+* `static/images/integrations/logos/mywebhook.png`: An image to represent
+  your integration in the user interface. Generally this Should be the logo of the
+  platform/server/product you are integrating. See [Documenting your
+  integration](#documenting-your-integration) for details.
+* `static/images/integrations/mywebbook/001.png`: A screen capture of your
+  integration for use in the user interface. You can add as many images as needed
+  to effectively document your webhook integration. See [Documenting your
+  integration](#documenting-your-integration) for details.
+* `zerver/fixtures/mywebhook/mywebhook_build.json`: Sample json payload data
+  used by tests. Can add multiple files. See [Testing and writing
+  tests](testing.html) for details.
+* `zerver/views/webhooks/mywebhook.py`: Includes the main webhook integration
+  function including any needed helper functions.
+
+### Walkthrough of `Hello World` webhook
+
+Below explains each part of a simple webhook integration, called **Hello
+World**. This webhook sends a "hello" message to the `test` stream and includes
+a link to the Wikipedia article of the day, which it formats from json data it
+receives in the http request.
+
+Use this walkthrough to learn how to write your first webhook
+integration.
+
+#### Step 1: Main webhook code
+
+The majority of the code for your webhook integration will be in a single
+python file in `zerver/views/webhooks/`. The Hello World integration is in
+`zerver/views/webhooks/helloworld.py`.
+
+```
+from __future__ import absolute_import
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+@api_key_only_webhook_view('HelloWorld')
+@has_request_variables
+def api_helloworld_webhook(request, user_profile, client,
+        payload=REQ(argument_type='body'), stream=REQ(default='test'),
+        topic=REQ(default='Hello World')):
+
+  # construct the body of the message
+  body = ('Hello! I am happy to be here! :smile: ')
+
+  # add a wikipedia link if there is one in the payload
+  if ('type' in payload and payload['type'] == 'wikipedia'):
+      body += '\nThe Wikipedia featured article for today is **[%s](%s)**' % (payload['featured_title'], payload['featured_url'])
```

Comment:
I think for the example we should have a `try/except KeyError` here to throw a `json_error` about the missing key, rather than having the server 500.  A good fraction of our webhooks do this.  

That said, it's possible we should figure out some clever framework way to do this automatically instead.